x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryStringFunction.java

```diff
@@ -61,27 +73,18 @@ public String functionName() {
     }
 
     @Override
-    public Query asQuery() {
-        Object queryAsObject = query().fold();
-        if (queryAsObject instanceof BytesRef queryAsBytesRef) {
-            return new QueryStringQuery(source(), queryAsBytesRef.utf8ToString(), Map.of(), null);
-        } else {
-            throw new IllegalArgumentException("Query in QSTR needs to be resolved to a string");
-        }
+    public Query asQuery(String queryText) {
+        return new QueryStringQuery(source(), queryText, Map.of(), null);
     }
 
     @Override
     public Expression replaceChildren(List<Expression> newChildren) {
-        return new QueryStringFunction(source(), newChildren.get(0));
+        return new QueryStringFunction(source(), newChildren.getFirst());
```

Comment:
Same as `Match` function, use `newChildren.get(position)` instead of `getFirst()`

Reply:
OK, done in 2ddcbb597956e1597964ba9b7003d581f73881f2